src/context.ts

```diff
@@ -865,6 +865,21 @@ export class Context<U extends Deunionize<tg.Update> = tg.Update> {
     })
   }
 
+  persistentChatAction(action: Shorthand<'sendChatAction'>[0], callback: () => Promise<void>, extra?: tt.ExtraSendChatAction) {
+    return new Promise<void>(async (resolve) => {
```

Comment:
The `new Promise` construction is unnecessary; make the function `async persistentChatAction (action: ...`